### ISSUE TITLE
After authentification it still wants to authentificate

### ISSUE THREAD

**Issue by s1de21** [vdesabou/alfred-spotify-mini-player#585]
i can't get it to work after the authentification i get the success screen and need to authetificate again...

**Comment by dr-code** [comment]
Im getting the same issue. I changed my default browser to chrome because the set up instructions said that it was not working with Safari. Despite this, I am still asked to authenticate again.

**Comment by vdesabou** [comment]
Are you both using Alfred 5 and latest version ?

**Comment by haffmaestro** [comment]
I have the same issue, I'm on 5.05. Using Chrome.

**Comment by dr-code** [comment]
I waited a day after authenticating with Chrome and it seems to have authenticated. I didn't do anything different except add myself as a user on the Spotify developer dashboard. Not sure if that had something to do with it. However it won't really play any music, but I'll have to create a new issue for that.

**Comment by pawlinsky** [comment]
I kinda fixed it with selecting kill the update library option from alfred. After 2 times, authentication worked.

**Comment by vdesabou** [comment]
Please try 12.1. I also had issues where settings could not be read/write successfully

**Comment by vdesabou** [comment]
Any feedback if 12.3 is resolving the issue ?

**Comment by haffmaestro** [comment]
It works for me! 🙏

**Comment by krishpop** [comment]
v12.3 actually broke things for me. I'm not sure if it's due to the fact that my spotify username (when I go to login) is different from the user ID I have set in the spotify player

**Comment by dr-code** [comment]
try adding the userID you have set in your spotify player to the dashboard
in the Spotify Developer page.

**Comment by AVE-PRCSS** [comment]
same issue here, tried everything but it won't go past the 'authenticate' process, if anyone find a way to fix this I would be super grateful

**Comment by vdesabou** [comment]
Can you explain what the issue is with screenshots for example

**Comment by AVE-PRCSS** [comment]
Of course ! So everything was going fine until the latest update got automatically installed. Right after that I got asked to ‘authenticate’ again (as shown on the screenshot below) and after rightly following the instructions given and despite the ’success’ screen that comes up after completion once I open back the mini player I get back to that screen that ask me to authenticate. I tried multiple times with different browser but nothing worked. I also checked that everything was in order regarding the Spotify mini app client ID, Secret,… and everything is fine on that part. I tried to link again the Spotify mini app scrupulously following the instructions, I also tried to uninstall/reinstall the workflow (the new one as well as the previous ones),… But nothing worked, no matter what I try I still end up stuck with this screen asking me to authenticate.

Spec : Mac OS monterey 12.6, Alfred 5, Spotify mini player workflow 12.3

**Comment by krishpop** [comment]
> try adding the userID you have set in your spotify player to the dashboard in the Spotify Developer page.

I tried doing this and it didn't help (also it only let me add my spotify user email there, not userID)

**Comment by vdesabou** [comment]
@AVE-PRCSS can you check in the new workflow configuration page that your client id and secret correctly appear there ?

**Comment by AVE-PRCSS** [comment]
@vdesabou yes I just checked again and they do correctly appear there. I also tried to generate a new secret but that didn't work either unfortunately

**Comment by vdesabou** [comment]
@AVE-PRCSS And what about those 2:

![CleanShot 2022-12-02 at 13 03 48@2x](https://user-images.githubusercontent.com/4061923/205288970-72034f0b-280c-4e9f-9774-f44429eb7ad7.jpg)

**Comment by AVE-PRCSS** [comment]
@vdesabou Ho my bad, I didn't knew/I totally forgot about those. I just checked and those are blank so I guess that is the issue. I also checked again the instructions given with the workflow as well as the spotify developer page but could't find any data corresponding this OAuth token, but seeing that it should be automatically set I guess the solution is trickier than that.

<img width="395" alt="Capture d’écran 2022-12-02 à 16 15 50" src="https://user-images.githubusercontent.com/113896504/205325482-d6cea77f-a2ae-4594-bc29-673c6473f694.png">

**Comment by vdesabou** [comment]
Yes it should be set automatically..When you authenticate, after the success screen (when Michael from The office is doing a party), the variables should be set automatically and appear in those boxes..

**Comment by AVE-PRCSS** [comment]
I see, so it's really weird that it does not happen that way when I authenticate. I guess the last resort I got to try is to delete my current spotify mini player app, then create a new one and follow the instructions from the start all over again. I'll let you know about the outcome of this procedure once it's done.

**Comment by krishpop** [comment]
Those fields did not populate for me either. Tried uninstalling and reinstalling and the issue is still there (also did not work with 12.2 this time)

**Comment by vdesabou** [comment]
This is what you should see when it works https://share.cleanshot.com/apbCG2

**Comment by krishpop** [comment]
Thanks! I updated Alfred, refreshed my client secret, switched to Chrome from Brave, turned off all privacy extensions and it worked (suspect DuckDuckGo extension may have been the culprit there).

**Comment by AVE-PRCSS** [comment]
I, for my part, unfortunately have to give up. I have literally tried everything, even deleting and creating a new spotify app on the dev page, and nothing worked out. I even tried with 5 different browsers with every extensions and security protocols  disabled, I always get to the same result as showed in the screenshots below. Even if I will miss this amazing workflow I can't afford to spend much more time on this, so I will just wait for the next update and pray that it will fix this issue. But anyway thanks for your time and your help !

<img width="1467" alt="Capture d’écran 2022-12-03 à 11 14 57" src="https://user-images.githubusercontent.com/113896504/205436307-d98a99dc-a9af-4aff-b4e2-4a13ae892f90.png">

<img width="1146" alt="Capture d’écran 2022-12-03 à 11 15 14" src="https://user-images.githubusercontent.com/113896504/205436309-1f2c2ae8-9c12-46a2-9c4e-0da79b4de3a9.png">

**Comment by vdesabou** [comment]
@AVE-PRCSS there is no reason to give up, we will figure it out !
Check if you can send a debug email as explained there https://alfred-spotify-mini-player.com/articles/support/

i can provide a beta version with more logs to understand where it does not work.
It sounds like a bug with Alfred itself, as it seems like environment variables are not being set up correctly..
Are you on latest Alfred 5 version and workflow 12.3 ?

**Comment by vdesabou** [comment]
FYI, I think this is a problem on Alfred side, so I reached to to the team to get their advices.

**Comment by vdesabou** [comment]
@AVE-PRCSS This is confirmed to be an issue on Alfred side, please use pre-release:

![CleanShot 2022-12-05 at 17 44 19](https://user-images.githubusercontent.com/4061923/205693525-a2aecf35-b245-435b-89b1-34c2b7259741.jpg)

**Comment by vdesabou** [comment]
closing as it is a bug in Alfred

**Comment by AVE-PRCSS** [comment]
Hello back, sorry for the late response and seeing that the case is closed it may not be necessary but I can now confirm that the workflow works like a charm again after updating to the latest version of Alfred !

**Comment by Abinthur** [comment]
identical case, identical solution with alfred pre-release 5.0.6. Thx!
